step1-04/demo/README.md

```diff
@@ -160,39 +159,37 @@ Now that we've got two Counters on our page, we can see that the string passed i
 />
 ```
 
-> Note that all non string values are passed through as JavaScript by wrapping it in `{}`
+> Note that all non-string values are passed through as JavaScript by wrapping them in `{}`.
 
-### Writing our Button Click
+### Writing our Button Click Handler
 
 Our next step is to wire up the button to increment the `counter` in our component state. This will very similar to what we did in step 3, but instead of placing the function in a script tag, we can create it as a class method, and keep it out of the global scope.
 
-> By convention we place methods below render, and private methods (those for internal use only) are prefixed with an underscore.
+> By convention we place other methods below render, and private methods (those for internal use only) are prefixed with an underscore.
+
+This function will update our component's state, incrementing the counter value by 1. (Note that `setState` only modifies the values of keys listed in object passed as its parameter.)
 
 ```jsx
 _onButtonClick = () => {
-  this.setState(prevState => ({ counter: prevState.counter + 1 }));
```

Comment:
this change might have crept in while you were doing edits. The functional state is prefered here.

Reply:
please revert this change as well